server/download.go

```diff
@@ -446,17 +447,17 @@ func watchDelta(ctx context.Context, g *LimitGroup, c *atomic.Int64, limit int64
 				continue
 			} else if maxDelta > 0 {
 				x := delta / maxDelta
-				if x < 1.2 {
+				if x < 1 {
 					continue
 				}
 
-				limit += int64(x)
+				limit += min(int64(x), int64(runtime.NumCPU()))
```

Comment:
`x` might be quite small here if it's slowy increasing, e.g. 1.2 which might round it down to 1 and never increase